### ISSUE TITLE
Bitbang is slow

### ISSUE THREAD

**Issue by RHSResearchLLC** [RHSResearchLLC/xvcd#2]
Currently running at ~300K Baud, due to data stall during write that appears to be inherent in the chip. Currently following up with FTDI and will update this issue as FTDI responds.

**Comment by swarren** [comment]
My guess: The FTDI is connected to the host via USB2. USB2 has an 8KHz frame rate, which translates to 125us per frame. Perhaps the xvcd protocol is performing "round trip" transactions (reading then writing then reading then writing, so waiting for a USB frame to complete each time), or simply isn't batching up enough write transactions at one go to stream multiple write packets to the device in one USB frame (I can't remember if multiple packets can be transferred to the device in a single frame in the USB protocol). So, what you're seeing is roughly 25us of time to write the data and then waiting 100us until the USB frame is over before the next write happens, repeatedly.

**Comment by swarren** [comment]
Oh, and the 64 comes from:

```
Bus 001 Device 002: ID 0403:6015 Future Technology Devices International, Ltd Bridge(I2C/SPI/UART/FIFO)
Device Descriptor:
  Configuration Descriptor:
    Interface Descriptor:
      Endpoint Descriptor:
        wMaxPacketSize     0x0040  1x 64 bytes
```

**Comment by makslevental** [comment]
Is there any update on this?

**Comment by RHSResearchLLC** [comment]
Not really- except that using OpenOCD is much faster than using Vivado to
program the flash
